src/Plugin/Sampler/MediaSource.php

```diff
@@ -80,19 +92,47 @@ public function collect() {
 
     foreach ($types as $name => $type) {
       if ($type->getEntityType()->getProvider() === 'media_entity') {
+        $source_configuration['source_field'] = $this->getSourceFieldIndex($type->getTypeConfiguration()['source_field'], $name);
```

Comment:
When I try on a project with contrib media, I get the following error:
`TypeError: Argument 1 passed to Drupal\sampler\Plugin\Sampler\MediaSource::getSourceFieldIndex() must be of the type string, null given, called in`

Maybe there is media type without source_field defined? 😬

Reply:
shit...